File: archives/leetcode2/0028. Implement strStr().py
class Solution:
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if not needle:
            return 0
        len_needle=len(needle)
        for k in range(len(haystack)-len_needle+1):
            if haystack[k:k+len_needle]==needle:
                return k
        return -1

File: archives/C#/0028. Implement strStr().cs
public class Solution {
    public int StrStr(string haystack, string needle) {
        int loc=-1;
        for(int i=0;i<haystack.Length-needle.Length+1;i++){
            if(checkIsSame(haystack,needle,i)){
                loc=i;
                break;
            }
        }
        return loc;
    }
    
    public bool checkIsSame(string haystack, string needle,int k){
        bool flag=true;
        for(int i=0;i<needle.Length;i++){
            if(needle[i]!=haystack[i+k]){
                flag=false;
                break;
            }
        }
        return flag;
    }
}
